### PR TITLE
Skip early and log when prefix mismatches

### LINKED ISSUE #88: prefix ignored when running
I have two calls for css files in my index.html, one uses the prefix the other does not. Both css files are picked up by sass-middleware instead of just the one with the prefix.

My config looks like this:
```
{
    src: path.join(__dirname, './client/sass'),
    dest: path.join(__dirname, './dist'),
    debug: true,
    outputStyle: 'compressed',
    prefix:  '/css'
}
```

I have links in my html that look like:
```
<link rel="stylesheet" href="/css/app.css" type="text/css">
<link rel="stylesheet" href="/outdated-browser/outdatedbrowser.min.css" type="text/css">
```

I am aware that I can just put the express.static method for outdated browser above the sass middleware and it won't get picked up by the middleware but the middleware shouldn't be picking it up anyway.

### PR DESCRIPTION
**Pull request by am11**
* If the requested file is not .css, log and skip (new: log, we were already skipping here)
* If the requested file does not starts with provided prefix, log and prefix (new)
* Provide reason for compile logs (new)
* Add more logger tests

Fixes #88

### CODE REVIEW

**Comment by am11** [comment]
@jacekkopecky, @silverlight513, can you please take a look?

File: middleware.js [hunk 1]
```diff
@@ -101,9 +107,7 @@ module.exports = function(options) {
 
     // This function will be called if something goes wrong
     var error = function(err, errorMessage) {
-      if (debug) {
-        log('error', 'error', errorMessage || err);
-      }
+      log('error', 'error', errorMessage || err);
```
Comment (am11) [inline middleware.js]: The severity level `error` should not be conditioned by `debug` check. Besides now the condition to check `debug` flag for severity debug in moved to the `log` function.

File: middleware.js [hunk 2]
```diff
@@ -285,15 +298,15 @@ module.exports = function(options) {
...
+          log('debug', 'compile', cssPath, 'was modified');
           return compile();
         }
 
         // Already compiled, check imports
         checkImports(sassPath, cssStats.mtime, function(changed) {
           if (debug && changed && changed.length) {
             changed.forEach(function(path) {
-              log('debug', 'modified import %s', path);
+              log('debug', 'compile', path, '(import file) was modified');
```
Comment (am11) [inline middleware.js]: Note that `%s` here was not expanding, now the fourth parameter is interpolated in the flattened log message.

File: middleware.js [hunk 3]
```diff
@@ -285,15 +298,15 @@ module.exports = function(options) {
         }
 
         if (sassStats.mtime > cssStats.mtime) { // Source has changed, compile it
-          if (debug) { log('debug', 'modified', cssPath); }
+          log('debug', 'compile', cssPath, 'was modified');
```
Comment (jacekkopecky) [inline middleware.js]: this implies the css was modified, not the scss; should the log have sassPath there instead?

**Comment by jacekkopecky** [comment]
This is very good, nicely consistent and much clearer, yes. Thanks for asking @am11

File: middleware.js (hunk 3 above)
Comment (am11) [inline middleware.js]: Thanks for catching this; this is fixed by bf8dfd9.

**Comment by am11** [comment]
Thanks @jacekkopecky for the review and remarks, much appreciated. The feedback has been addressed.

**Comment by am11** [comment]
[<kbd>v0.11.0</kbd>](https://github.com/sass/node-sass-middleware/releases/tag/v0.11.0) has been released.

**Comment by jacekkopecky** [comment]
Happy to help, cheers!

**Pull request merged**